libvcx/src/api_vcx/api_handle/mediated_connection.rs

```diff
@@ -220,48 +221,39 @@ pub async fn handle_message(handle: u32, message: &str) -> LibvcxResult<u32> {
     })?;
     let profile = get_main_profile_optional_pool(); // do not throw if pool is not open
     connection.handle_message(message, &profile).await?;
-    CONNECTION_MAP.insert(handle, connection)?;
-    Ok(error::SUCCESS_ERR_CODE)
+    CONNECTION_MAP.insert(handle, connection)
 }
 
 pub async fn update_state(handle: u32) -> LibvcxResult<u32> {
     let mut connection = CONNECTION_MAP.get_cloned(handle)?;
-    let res = if connection.is_in_final_state() {
+    if connection.is_in_final_state() {
         info!(
             "connection::update_state >> connection {} is in final state, trying to respond to messages",
             handle
         );
         let profile = get_main_profile_optional_pool(); // do not throw if pool is not open
-        match connection
+        connection
             .find_and_handle_message(&profile, &get_main_agency_client().unwrap())
```

Comment:
unrelated to this PR, but is there any issues with these `unwrap`s? i.e. is panicking dangerous for the iOS/android/node consumers?

Reply:
Good catch - there should be no `unwraps` anywhere in the code (with exception of test code). The unwrap() on the line (and few similar I found now in `api_handle` module) are unjustified. 

I am not sure if perhaps different wrappers react differently on panic. Afaik in the old (current) node wrapper it results in exception on nodejs. But it can have more consequences, if you happen to own a mutes/rwlock lock while panicking, it can cause issues with the subsequent calls as the lock become poisoned. 

So definitely we want to avoid any panic, generally the application should shut down on panic, similarly to `Error` in Java